[DOCS] Add cpx mode (#1555)
ROCm/rccl: CHANGELOG.md

  environment variable `RCCL_DISABLE_RAIL_TREES=1`.
* Additional debug information about how the trees are built can be logged to the GRAPH logging subsys by setting
  `RCCL_OUTPUT_TREES=1`.

> * Extended the documentation with NPS4 and CPX partition modes performance benefits on MI300X.

Reviewer: ```suggestion
* Added documentation about the NPS4 and CPX partition modes performance benefits on the MI300X.
```
